lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart

```diff
@@ -29,85 +52,213 @@ class _NavigationBarHomeScreenState extends State<NavigationBarHomeScreen> {
+        analyticsEvent: (int index) => _analyticsEvents[index],
+      ),
+      body: AnimatedSwitcher(
+        duration: const Duration(milliseconds: 300),
+        child: AnimatedSwitcher(
+          duration: const Duration(milliseconds: 300),
+          switchInCurve: Curves.easeInOutQuart,
+          switchOutCurve: Curves.easeInOutQuart,
+          transitionBuilder: (child, animation) {
+            return SlideTransition(
+              position: Tween<Offset>(
+                begin: const Offset(0, 0.3),
+                end: Offset.zero,
+              ).animate(animation),
+              child: child,
+            );
           },
+          child: <Widget>[
+            const ProfileSelectionScreen(),
+            const FamilyOverviewPage(),
+            const USPersonalInfoEditPage(),
+          ][_currentIndex],
         ),
       ),
-      body: BaseStateConsumer(
-        cubit: _cubit,
-        onCustom: _handleCustom,
-        onInitial: (context) {
-          return AnimatedSwitcher(
-            duration: const Duration(milliseconds: 300),
-            child: AnimatedSwitcher(
-              duration: const Duration(milliseconds: 300),
-              switchInCurve: Curves.easeInOutQuart,
-              switchOutCurve: Curves.easeInOutQuart,
-              transitionBuilder: (child, animation) {
-                return SlideTransition(
-                  position: Tween<Offset>(
-                    begin: const Offset(0, 0.3),
-                    end: Offset.zero,
-                  ).animate(animation),
-                  child: child,
-                );
-              },
-              child: <Widget>[
-                const ProfileSelectionScreen(),
-                const FamilyOverviewPage(),
-                const USPersonalInfoEditPage(),
-              ][_currentIndex],
-            ),
-          );
-        },
-      ),
     );
   }
 
+  Future<void> _onDestinationSelected(
+    int index, {
+    NavigationBarHomeScreenUIModel? uiModel,
+  }) async {
+    unawaited(SystemSound.play(SystemSoundType.click));
+    unawaited(HapticFeedback.selectionClick());
+    if (index == NavigationBarHomeScreen.homeIndex) {
+      _setIndex(index);
+    } else {
+      await FamilyAuthUtils.authenticateUser(
+        context,
+        checkAuthRequest: CheckAuthRequest(
+          navigate: (context, {isUSUser}) async {
+            if (index == NavigationBarHomeScreen.familyIndex &&
+                true == uiModel?.cachedMembers?.isNotEmpty) {
+              _showCachedMembersDialog(context, uiModel!.cachedMembers!);
+            } else {
+              _setIndex(index);
+            }
+          },
+        ),
+      );
+    }
+  }
+
+  void _setIndex(int index) {
+    setState(() {
+      _currentIndex = index;
+    });
+  }
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    _cubit.onDidChangeDependencies();
+    _cubit.init();
+  }
```

Comment:
If we replace the ProfileSelectionScreen, we are missing the AppThemeSwitcher, should this be here, or on the newstyle FamilyHomeScreen? I would expected it here

Reply:
Good catch, it should be here! Will add it.